Miscellaneous WDL updates

aofarrel opened the pull request:
I've been doing a lot of WDL lately, and keep referring back to this document as boilerplate code. So I've got a vested interest in making sure it's right!

* Moves some of the Terra specific stuff to the Terra document
  * This also updates some of the stuff in said Terra document
* Fixes a typo in a recipe where the same file was getting size()ed twice
* Add explainer for size() on optional files, because I keep getting that syntax mixed up. It seems that it really has to be select_first([ceil(size(optional_file)), 0]) and not ceil(size(select_first([optional file, 0]))), which I guess kind of makes sense...?

aofarrel committed:
fix typo in recipe

aofarrel committed:
Add info on sizing optional files + x-refs

aofarrel committed:
Remove some redundancies with Terra document

aofarrel committed:
rm bash comment from Terra doc, rm Firecloud API stuff from main doc

aofarrel committed:
Update Terra doc TOC

bethsheets reviewed (approved):
LGTM. Thanks, @aofarrel! I don't see the content in the Terra-specific doc added yet -- I'm guessing that will be another PR?

aofarrel committed:
Link to WDL style guide, admit I disobey it

Add cheat sheet, fix indents

Revert "Add cheat sheet, fix indents"

This reverts commit 804c76b478c4c3dd2d363fb486d39b5f75c70e95.

Revert "Link to WDL style guide, admit I disobey it"

This reverts commit aaee0bd2576c242fa8467be50b01e07237aa4a6d.

aofarrel commented:
You can safely ignore [a02f5c3](https://github.com/ucsc-cgp/training-resources/pull/6/commits/a02f5c325b322de2f684b3b65c6c0d0ccbd9c1cc) and the force-push; had to do some git magic as I accidentally added additional commits after Beth's review. Those commits are now on a new branch.

aofarrel commented:
> LGTM. Thanks, @aofarrel! I don't see the content in the Terra-specific doc added yet -- I'm guessing that will be another PR?

The Terra-specific document was basically an older version of some of the content in the overall document. So, it already had a lot of information that was either verbatim from what was in the main readme.md, or just worded differently. That's not to say nothing changed -- some of the stuff in the main document were newer and better written, and there's also the new addition of the section about using firecloud API to get better error output.

There was a handful of things where I used the old content already in the Terra-specific document instead of replacing it with the new content, because the old content was less verbose.
